File: 7.dn架构师/code/lsn17/DN_ALAN_16/app/src/main/java/com/dn_alan/myapplication/MainActivity.java
package com.dn_alan.myapplication;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;

import com.dn_alan.router_annotation.Extra;
import com.dn_alan.router_annotation.Route;

import java.util.HashMap;
import java.util.Map;

@Route(path = "/main/test")
public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


//        Map<String, String> map = new HashMap<>();
//        DNRouter$$Group$$main dnRouter$$Group$$main = new DNRouter$$Group$$main();
//        dnRouter$$Group$$main.loadInto(map);
//        Intent intent = new Intent(this, map.get(path).class);
//        startActivity(intent);
    }
}

File: recyclerview/src/main/java/com/east/recyclerview/MainActivity.kt
package com.east.recyclerview

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.east.recyclerview.itemanimator.ItemAnimatorActivity
import com.east.recyclerview.itemdecoration.ItemDecorationActivity
import com.east.recyclerview.recycleranalysis.RecyclerAnalysisActivity
import com.east.recyclerview.swap.SwapActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun onClick(v:View){
        when(v){
            //
            itemDecoration -> startActivity(Intent(this,ItemDecorationActivity::class.java))
            //动画
            animator -> startActivity(Intent(this,ItemAnimatorActivity::class.java))
            //交换位置
            swap -> startActivity(Intent(this,SwapActivity::class.java))
            //辉哥的RecyclerView封装
            btn_hui -> startActivity(Intent(this,RecyclerAnalysisActivity::class.java))
        }
    }
}
